README.md

```diff
@@ -44,196 +48,396 @@ AgentsClientSDK.xcframework/
+        }
+    }
+   ```
+
+2. **Initialize SDK**:
+   ```swift
+   private func initializeSDK() {
+       guard let appSettings = self.appSettings else { return }
+        // Check if SDK is already initialized
+        if !AgentsClientSdk.shared.isInitialized {
+            Task {
+                do {
+                    self.client = try await AgentsClientSdk.shared.initSDK(
+                        authenticationDelegate: self,
+                        appSettings: appSettings
+                    )
+                    // If client is still nil after initialization, try to get it from shared SDK
+                    if self.client == nil {
+                        self.client = AgentsClientSdk.shared.client
+                    }
+                } catch let error as SDKError {
+                    let errorMsg =
+                        "\(error.errorCode): \(error.localizedDescription)"
+                    print("ContentView Error: \(errorMsg)")
+                    await MainActor.run {
+                        print(errorMsg)
+                    }
+                } catch {
+                    print("ContentView Error: \(error)")
+                    await MainActor.run {
+                        print(
+                            "Initialization failed: \(error.localizedDescription)"
+                        )
+                    }
+                }
+            }
+        }
+   }
+   ```
+
+3. **Wait for Initialization**:
+   ```swift
+   private func waitforInitialization() async {
+       while self.client?.isInitialized != true {
+            try? await Task.sleep(nanoseconds: 100_000_000)  // 100ms
+
+            // Safety check to prevent infinite loop
+            if client == nil {
+                print("Client became nil, stopping wait loop")
+                break
+            }
+        }
 
-![RedirectURI](assets/RedirectURI.png)
+        // Update the local state when initialization is complete
+        if client?.isInitialized == true {
+            await MainActor.run {
+                self.isInitialized = true
+            }
+        }
+   }
+   ```
+4. **Initialize SDK**:
+   ```swift
+   .onAppear {
+            self.appSettings = loadAppSettings()
+            initializeSDK()
+        }
+   ```
 
+5. **Monitor Initialization State**:
+   ```swift
+   .onChange(of: client?.isInitialized) { oldValue, newValue in
+       switch newValue {
+       case .some(true):
+           isSpeechEnabled = AgentsClientSdk.shared.isSpeechEnabled
+           hideSignInContent()
+       case .some(false):
+           Task { await waitforInitialization() }
+       case .none:
+           print("Client is nil")
+       }
+   }
+   ```
 
+### 5. Chat Component Implementation (PluggableChatComponent.swift)
 
-### Step 4: Initialize the SDK Connection in Your App
+Created a fully-featured, pluggable chat component with: Please refer samples/TextWithAdaptiveCardClientApp/TextWithAdaptiveCardClientApp/PluggableChatComponent.swift
```

Comment:
can be just what they need to do

We have a pluggable component that can be directly integrated into apps. No need to build a ChatUI from scratch.
For more details, check doc here: